### PR TITLE
Front/details

### PR DESCRIPTION
**Pull request by andres-rodriguez16**
updating styles of the detail of each video game

### CODE REVIEW

File: api/src/routes/rutaPorId.js [hunk 1]
```diff
@@ -11,13 +11,42 @@ router.get("/:idVi", async (req, res) => {
...
       const result = await getGameId(idVi)
+      
       res.json(result) 
     } 
    else {
       const busquefaDB = await Videogame.findByPk(idVi,{
         include : Genero
       })
-      res.send(busquefaDB)
+      // console.log(busquefaDB.dataValues.Generos.map(el => el.name));
```
Comment (nelsoncardenas) [inline api/src/routes/rutaPorId.js]: Remove comment
```suggestion
```

File: api/src/utils/busquedaPorID.js [hunk 2]
```diff
@@ -5,20 +5,41 @@ async function getGameId(idVi) {
...
+      name: searchPorId.data.name,
+      description: searchPorId.data.description_raw,
+      img: searchPorId.data.background_image,
+      rating: searchPorId.data.rating,
+      released: searchPorId.data.released,
+      Generos: searchPorId.data.genres?.map(g => g.name),
+      platforms: searchPorId.data.platforms?.map(p => p.platform.name)
+    };
+    let genres = ""
+    game.Generos.forEach((el, i) => {
```
Comment (nelsoncardenas) [inline api/src/utils/busquedaPorID.js]: you should avoid using names like `el`. Try to use meaningful names.

File: api/src/utils/busquedaPorID.js [hunk 3]
```diff
@@ -5,20 +5,41 @@ async function getGameId(idVi) {
...
+      if (i === game.Generos.length - 1) {
+        genres = genres + " " + el
+        return;
+      }
+      genres = genres + ` ${el} -`
+    });
+    game.Generos = genres
+
+    let platform = ""
+    game.platforms.forEach((el, i) => {
```
Comment (nelsoncardenas) [inline api/src/utils/busquedaPorID.js]: change `el` name

File: api/src/utils/fecthAPIGamesget.js [hunk 4]
```diff
@@ -14,6 +14,8 @@ async function fecthAPIGames(url, count) {
 
 function datosTraidosDeLaApi(datos) {
   try {
+    // datos.genres.replace("Action", "hola")
```
Comment (nelsoncardenas) [inline api/src/utils/fecthAPIGamesget.js]: Remove unnecesary comment

File: client/src/App.module.css [hunk 5]
```diff
@@ -15,4 +15,25 @@
...
 }
+@media (max-width: 600px) {
+  .background {
+    background-color: hsla(0, 100%, 50%, 1);
+    background-image: url('./img/gradient_mesh.png');
+    min-height: 100vh;
+    height: auto;
+    display: flex;
+    justify-content: center;
+    /* align-items: center; */
```
Comment (nelsoncardenas) [inline client/src/App.module.css]: another comment

File: client/src/componets/Details/Details.module.css [hunk 6]
```diff
@@ -1,66 +1,141 @@
...
-.lanzamiento{
+
+.rating {
   font-family: 'Montserrat', sans-serif;
-  padding-top: 20px;
-  text-align:center;
+  color: #FE7F3C;
 }
+
+.lanzamiento {
```
Comment (nelsoncardenas) [inline client/src/componets/Details/Details.module.css]: You are not using english. Maybe you can create an issue for this one instead of changing it for this PR.

File: client/src/componets/Details/Details.module.css [hunk 7]
```diff
@@ -1,66 +1,141 @@
...
+  .img {
+    height: 200px;
+    border-radius: 10px;
+    width: 250px;
+    margin-left: 30px;
+    margin-right: 30px;
+    margin-top: 0;
+    margin-bottom: 0;
+  }
+
```
Comment (nelsoncardenas) [inline client/src/componets/Details/Details.module.css:93]: Unnecessary space.

File: client/src/componets/Formulario/Plataformas.js [hunk 8]
```diff
@@ -18,4 +18,10 @@ const platforms = [
   'PlayStation 2',
   "Wii U"
 ] 
+// {detailsVideoGames.platforms.splice(detailsVideoGames.platforms.indexOf('Nintendo Switch'),detailsVideoGames.platforms.indexOf('Nintendo Switch'), "https://cdn-icons-png.flaticon.com/512/871/871377.png" )}
```
Comment (nelsoncardenas) [inline client/src/componets/Formulario/Plataformas.js]: Why did you add those comments?

File: client/src/componets/Navbar/Navbar.jsx [hunk 9]
```diff
@@ -20,7 +20,7 @@ const NavBar = () => {
         </Link>
         <section className={s['navbar--section']}>
           <ul className={s['navbar--section__container']}>
-            <li>
+            <li >
```
Comment (nelsoncardenas) [inline client/src/componets/Navbar/Navbar.jsx]: Unnecessary extra space

File: client/src/componets/filters/Filters.jsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+  function handleFiltroPorGenro(e) {
+    dispatch(filtroPorGenero(e.target.value));
+    setActualPage(1);
+  }
+
+  function handleFiltroPorRating(e) {
+    dispatch(ratingSort(e.target.value));
+    setActualPage(1)
+  }
+  function handleFiltroPorAlpha(e) {
```
Comment (nelsoncardenas) [inline client/src/componets/filters/Filters.jsx:32]: add break line

File: client/src/componets/filters/Filters.jsx [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    dispatch(filtroPorAlpha(e.target.value));
+    setActualPage(1)
+  }
```
Comment (nelsoncardenas) [inline client/src/componets/filters/Filters.jsx:35]: add break line

**Review by nelsoncardenas — CHANGES_REQUESTED** [review]

**Comment by andres-rodriguez16** [comment]
recommendations corrected, ready to revisit

File: api/src/app.js [hunk 12]
```diff
@@ -9,12 +9,13 @@ require('./db.js');
 const server = express();
 
 server.name = 'API';
+// 'http://localhost:3000'
```
Comment (yulianacastrodiaz) [inline api/src/app.js]: remove comment

File: api/src/db.js [hunk 13]
```diff
@@ -5,13 +5,38 @@ const path = require('path');
...
+    password: PGPASSWORD,
+    pool: {
+      max: 3,
+      min: 1,
+      idle: 10000,
+    },
+    dialectOptions: {
+      ssl: {
+        require: true,
+        // Ref.: https://github.com/brianc/node-postgres/issues/2009
```
Comment (yulianacastrodiaz) [inline api/src/db.js]: remove comment

File: api/src/db.js [hunk 14]
```diff
@@ -5,13 +5,38 @@ const path = require('path');
...
+        rejectUnauthorized: false,
+      },
+      keepAlive: true,
+    },
+    ssl: true,
+  })
+    :
+    new Sequelize(`postgres://${DB_USER}:${DB_PASSWORD}@${DB_HOST}/videogames`, {
+      logging: false, // set to console.log to see the raw SQL queries
```
Comment (yulianacastrodiaz) [inline api/src/db.js]: idem

File: api/src/db.js [hunk 15]
```diff
@@ -5,13 +5,38 @@ const path = require('path');
...
+      native: false, // lets Sequelize know we can use pg-native for ~30% more speed
```
Comment (yulianacastrodiaz) [inline api/src/db.js]: idem

File: api/src/utils/busquedaPorID.js [hunk 16]
```diff
@@ -5,20 +5,41 @@ async function getGameId(idVi) {
...
+  
+    
+    return game;
   } catch (error) {
     return error;
   }
-  
+
 }
 module.exports = { getGameId };
```
Comment (yulianacastrodiaz) [inline api/src/utils/busquedaPorID.js]: missing break line, please review code quality

File: api/src/utils/fecthAPIGamesget.js (hunk 4 above)
Comment (yulianacastrodiaz) [reply api/src/utils/fecthAPIGamesget.js]: remove comment, this conversation not resolved

File: client/src/App.module.css (hunk 5 above)
Comment (yulianacastrodiaz) [reply client/src/App.module.css]: idem

File: client/src/App.module.css [hunk 17]
```diff
@@ -15,4 +15,25 @@
...
+  
+  .transparency {
+    background: rgba(255, 255, 255, 0.85);
+    backdrop-filter: blur(30px);
+    border-radius: 52px;
+    height: 85%;
+    width: 80%;
+    margin: 50px 50px;
+  }
+}
```
Comment (yulianacastrodiaz) [inline client/src/App.module.css:38]: missing break line, please review the code quality

File: client/src/componets/Details/Details.module.css [hunk 18]
```diff
@@ -1,66 +1,132 @@
...
+  select,
+  ul {
+    list-style: none
+  }
+
+  .platforms {
+    font-family: 'Montserrat', sans-serif;
+  }
 }
```
Comment (yulianacastrodiaz) [inline client/src/componets/Details/Details.module.css]: missing break line, please review the code quality

File: client/src/componets/Formulario/Formulario.module.css [hunk 19]
```diff
@@ -25,29 +15,40 @@
...
+  font-family: 'Montserrat', sans-serif;
 }
 
 .button__submit {
 
-  background: #fff; /* color de fondo */
-  color:  #FE7F3C; /* color de fuente */
-  border: 2px solid  #FE7F3C; /* tamaño y color de borde */
+  background: #fff;
+  /* color de fondo */
```
Comment (yulianacastrodiaz) [inline client/src/componets/Formulario/Formulario.module.css]: remove comment

File: client/src/componets/Formulario/Formulario.module.css [hunk 20]
```diff
@@ -25,29 +15,40 @@
...
+  color: #FE7F3C;
+  /* color de fuente */
```
Comment (yulianacastrodiaz) [inline client/src/componets/Formulario/Formulario.module.css]: idem

File: client/src/componets/Formulario/Formulario.module.css [hunk 21]
```diff
@@ -25,29 +15,40 @@
...
+  border: 2px solid #FE7F3C;
+  /* tamaño y color de borde */
```
Comment (yulianacastrodiaz) [inline client/src/componets/Formulario/Formulario.module.css]: idem

File: client/src/componets/Formulario/Formulario.module.css [hunk 22]
```diff
@@ -25,29 +15,40 @@
...
   border-radius: 50px;
   padding: 16px 40px;
-  border-radius: 3px; /* redondear bordes */
+  border-radius: 3px;
+  /* redondear bordes */
```
Comment (yulianacastrodiaz) [inline client/src/componets/Formulario/Formulario.module.css]: idem

File: client/src/componets/Formulario/Formulario.module.css [hunk 23]
```diff
@@ -25,29 +15,40 @@
...
   position: relative;
   z-index: 1;
   overflow: hidden;
-  margin-left: 40%;
+  /* margin-left: 40%; */
```
Comment (yulianacastrodiaz) [inline client/src/componets/Formulario/Formulario.module.css]: idem

File: client/src/componets/Formulario/Formulario.module.css [hunk 24]
```diff
@@ -25,29 +15,40 @@
...
+  display: block;
+  margin: 0px auto;
+  font-family: 'Montserrat', sans-serif;
 }
 
-.button__submit:hover{
-  color: #fff; /* color de fuente hover */
+.button__submit:hover {
+  color: #fff;
+  /* color de fuente hover */
```
Comment (yulianacastrodiaz) [inline client/src/componets/Formulario/Formulario.module.css]: idem

File: client/src/componets/Formulario/Formulario.module.css [hunk 25]
```diff
@@ -25,29 +15,40 @@
...
 }
+
 .button__submit::after {
   content: "";
-  background:  #FE7F3C; /* color de fondo hover */
+  background: #FE7F3C;
+  /* color de fondo hover */
```
Comment (yulianacastrodiaz) [inline client/src/componets/Formulario/Formulario.module.css]: idem

File: client/src/componets/Navbar/Navbar.module.css [hunk 26]
```diff
@@ -52,35 +52,58 @@ a:focus {
...
+    max-width: 40%;
+    padding-right: 50px;
+    margin-right: 10%;
+    margin-top: 20px;
+  }
+  .navbar--section li {
+   margin-right: 10px;
+   margin-left: 10px;
+  }
+}
```
Comment (yulianacastrodiaz) [inline client/src/componets/Navbar/Navbar.module.css]: missing break line, please review the code quality

File: client/src/redux/actions/action.js [hunk 27]
```diff
@@ -14,115 +14,117 @@ export const ORDEN_FECHA = "ORDEN_FECHA"
...
 
-export function ordenFecha (orden){
+export function ordenFecha(orden) {
   return {
-    type : ORDEN_FECHA,
-    payload : orden
+    type: ORDEN_FECHA,
+    payload: orden
   }
 }
```
Comment (yulianacastrodiaz) [inline client/src/redux/actions/action.js]: missing break line, please review the code quality

**Review by yulianacastrodiaz — CHANGES_REQUESTED** [review]

**Comment by andres-rodriguez16** [comment]
recommendations checklist

File: api/src/routes/rutaPorGames.js [hunk 28]
```diff
@@ -80,37 +78,35 @@ router.post("/", async (req, res) => {
...
+      {
+        where: {
+          id: id
+        }
+      })
+    res.send("videogames actualizado")
+  } catch (error) {
+  }
 })
 module.exports = router;
```
Comment (yulianacastrodiaz) [inline api/src/routes/rutaPorGames.js]: missing break line

File: client/src/componets/Formulario/Formulario.module.css [hunk 29]
```diff
@@ -109,12 +116,62 @@
...
+  border: 1px solid  #08caec;
+}
+@media (max-width: 600px) {
+  .h1__create {
+    color: #FE7F3C;
+    text-align: center;
+    margin-top: 20px;
+    margin-bottom: -20px;
+  }
 }
```
Comment (yulianacastrodiaz) [inline client/src/componets/Formulario/Formulario.module.css]: missing break line

File: client/src/componets/Navbar/Navbar.module.css [hunk 30]
```diff
@@ -52,35 +52,56 @@ a:focus {
...
+    padding-right: 50px;
+    margin-right: 10%;
+    margin-top: 20px;
+  }
+
+  .navbar--section li {
+    margin-right: 10px;
+    margin-left: 10px;
+  }
+};
```
Comment (yulianacastrodiaz) [inline client/src/componets/Navbar/Navbar.module.css]: missing break line

File: client/src/redux/actions/action.js [hunk 31]
```diff
@@ -14,115 +14,117 @@ export const ORDEN_FECHA = "ORDEN_FECHA"
...
+        const response = await axios.get(`/videogame/${id}`)
         dispatch({
           type: GET_VIDEOGAMES_POR_ID,
-          payload: response
+          payload: response.data
         })
-      })
-      .catch(e => console.log(e))
+      } catch (error) {
+        console.log(error);
```
Comment (yulianacastrodiaz) [inline client/src/redux/actions/action.js]: esto no debería tener un console.log, debe lanzar el error.

File: client/src/redux/actions/action.js [hunk 32]
```diff
@@ -14,115 +14,117 @@ export const ORDEN_FECHA = "ORDEN_FECHA"
...
+      const response = await axios.get(`/genres/`)
         dispatch({
           type: GET_GENRES,
-          payload: response,
+          payload: response.data,
         })
-      })
-      .catch(e => console.log(e))
+    } catch (error) {
+      console.log(error)
```
Comment (yulianacastrodiaz) [inline client/src/redux/actions/action.js]: idem

File: client/src/redux/actions/action.js [hunk 33]
```diff
@@ -14,115 +14,117 @@ export const ORDEN_FECHA = "ORDEN_FECHA"
...
+export function ordenFecha(orden) {
   return {
-    type : ORDEN_FECHA,
-    payload : orden
+    type: ORDEN_FECHA,
+    payload: orden
   }
-}
\ No newline at end of file
+};
```
Comment (yulianacastrodiaz) [inline client/src/redux/actions/action.js:127]: missing break line

**Review by yulianacastrodiaz — CHANGES_REQUESTED** [review]

File: api/src/routes/rutaPorId.js [hunk 34]
```diff
@@ -11,13 +11,41 @@ router.get("/:idVi", async (req, res) => {
...
+        return;
+      }
+      platform = platform + `  ${platform} -`
+    });
+    videoGame.platforms = platformFinal
+    videoGame.Generos = strFinal
+      res.send(videoGame)
     }
   } catch (error) {
     console.log(error)
```
Comment (yulianacastrodiaz) [inline api/src/routes/rutaPorId.js]: el catch, debe lanzar un error, no hacerle console.log

File: api/src/utils/busquedaPorID.js [hunk 35]
```diff
@@ -5,20 +5,39 @@ async function getGameId(idVi) {
...
+      if (i === game.platforms.length - 1) {
+        platforms = platforms + " " + platform
+        return;
+      }
+      platforms = platforms + `  ${platform} -`
+    });
+    game.platforms = platforms
+    return game;
   } catch (error) {
     return error;
```
Comment (yulianacastrodiaz) [inline api/src/utils/busquedaPorID.js]: el catch lanza un error, no lo retorna, lo lanza

File: client/src/redux/actions/action.js [hunk 36]
```diff
@@ -14,115 +14,117 @@ export const ORDEN_FECHA = "ORDEN_FECHA"
...
-      .catch(e => console.log(e))
+  return async function (dispatch) {
+    try {
+      const response = await axios.get(`/videogames/`)
+      dispatch({
+        type: GET_VIDEOGAMES,
+        payload: response.data
+      });
+    } catch (error) {
+      console.log(error);
```
Comment (yulianacastrodiaz) [inline client/src/redux/actions/action.js]: controlar el error

File: client/src/redux/actions/action.js [hunk 37]
```diff
@@ -14,115 +14,117 @@ export const ORDEN_FECHA = "ORDEN_FECHA"
...
+        const response = await axios.get(`/videogame/${id}`)
         dispatch({
           type: GET_VIDEOGAMES_POR_ID,
-          payload: response
+          payload: response.data
         })
-      })
-      .catch(e => console.log(e))
+      } catch (error) {
+        return error;
```
Comment (yulianacastrodiaz) [inline client/src/redux/actions/action.js]: controlar el error, no retornarlo

**Review by yulianacastrodiaz — APPROVED** [review]

File: api/src/routes/rutaPorGames.js [hunk 38]
```diff
@@ -80,37 +78,35 @@ router.post("/", async (req, res) => {
...
+      name: name,
+      rating: rating
+    },
+      {
+        where: {
+          id: id
+        }
+      })
+    res.send("videogames actualizado")
+  } catch (error) {
```
Comment (yulianacastrodiaz) [inline api/src/routes/rutaPorGames.js:109]: controlar el error

**Review by yulianacastrodiaz — CHANGES_REQUESTED** [review]

File: client/src/componets/Details/Details.module.css (hunk 6 above)
Comment (nelsoncardenas) [reply client/src/componets/Details/Details.module.css]: @andresyrg16  I'm still waiting for the issue creation

**Comment by nelsoncardenas** [comment]
I noticed you have the path `client/src/componets/filters/Filters.jsx`, but looking at the components folder, I can see you use Capitalized folders:

image

So I think maybe you have two errors in two folders. `landingPage` and `filters`. What do you think?

File: client/src/componets/filters/Filters.jsx (hunk 10 above)
Comment (nelsoncardenas) [reply client/src/componets/filters/Filters.jsx:32]: You didn't solve the problem. Evidence:
image

File: client/src/componets/filters/Filters.jsx (hunk 11 above)
Comment (nelsoncardenas) [reply client/src/componets/filters/Filters.jsx:35]: You didn't solve the problem. Evidence:
image

**Comment by nelsoncardenas** [comment]
Are filters a component or a helper?

File: client/src/componets/Paginado/Paginado.module.css [hunk 39]
```diff
@@ -38,8 +38,3 @@
   color: rgb(255, 255, 255);
 }
 
-
-/* .paginado button:hover {
-  background-color: rgb(255, 255, 255);
-  color: #FE7F3C
-} */
```
Comment (nelsoncardenas) [inline client/src/componets/Paginado/Paginado.module.css:45]: Missing break line

**Review by nelsoncardenas — CHANGES_REQUESTED** [review]

**Review by nelsoncardenas — APPROVED** [review]

**Review by yulianacastrodiaz — APPROVED** [review]
